velox/functions/prestosql/aggregates/PrestoHasher.cpp

```diff
@@ -237,6 +238,14 @@ void PrestoHasher::hash<TypeKind::ROW>(
   auto rawHashes = hashes->asMutable<int64_t>();
   auto rowChildHashes = childHashes->as<int64_t>();
 
+  if (isTimestampWithTimeZoneType(vector_->base()->type())) {
+    // Hash only timestamp value.
```

Comment:
This looks like a gap in Presto's checksum implementation. The checksum would be the same even if data is very different, e.g. have same timestamps but different time zones. Let's bring it up with Presto folks.

Reply:
I have raised an issue here : https://github.com/prestodb/presto/issues/17274